### ISSUE TITLE
The bug in the code of initation of draft tokens

### ISSUE THREAD

**Issue by Haodong-Lei-Ray** [junhyukso/GSD#1]
Hi.

I am grateful for your work, which is a big contribution to Speculative sampling.

But I still have some questions about the code.

For example, if the code of initiation of draft tokens is like that:

https://github.com/junhyukso/GSD/blob/4216c7c2bd5d04be12410adb07143af4cd7f7b88/scheduler/jacobi_iteration_lumina_mgpt_sjd.py#L570-L573

But the process of sampling tokens deals with probability like that:

https://github.com/junhyukso/GSD/blob/4216c7c2bd5d04be12410adb07143af4cd7f7b88/scheduler/logit_processor_3dim.py#L67-L73

https://github.com/junhyukso/GSD/blob/4216c7c2bd5d04be12410adb07143af4cd7f7b88/scheduler/logit_processor_3dim.py#L127-L131

Won't it let all probability == -inf, so that exists a bug?

**Comment by junhyukso** [comment]
Hello, thanks for raising the issue. Could you please clarify what the bug is?

It seems there's  slight bug if `rand_token` is generated from `0 ~ img_vocab_size`, while `suppress_token_mask` is from `4 ~ 8195`.  (meaning that last four tokens are never be initial draft.)

**Comment by Haodong-Lei-Ray** [comment]
Hello.

For example, if the ```rand_tokens``` = [1, 1, 1] and their scores = [ [0, 1, ..., 0], [ 0, 1, ..., 0],[ 0, 1, ..., 0],[ 0, 1, ..., 0] ] 
(the logits = [ [-inf, 0, ..., -inf], [-inf, 0, ..., -inf],[-inf, 0, ..., -inf],[-inf, 0, ..., -inf] ]), their logits will be processed to [ [-inf, -inf, ..., -inf], [-inf, -inf, ..., -inf],[-inf, -inf, ..., -inf],[-inf, -inf, ..., -inf] ] by the logit_processor because  ```self.suppress_token_mask``` only allows [   4,    5,    6,  ..., 8193, 8194, 8195] do not equal -inf. Therefore, the final scores transformed by logits will be wrong.

Well, I find this bug is a low-probability event.
